### PR TITLE
Create reset password API endpoint

### PR DESCRIPTION
**Pull request by its-edmund**
## PR Title/Tagline

Issue Number(s): #number_of_issue. 

What does this PR change and why?

### Checklist

- [x]  Database schema docs have been updated or are not necessary
- [x]  Code follows design and style guidelines
- [ ]  Code is commented with doc blocks
- [ ]  Tests have been written and executed or are not necessary
- [ ]  Latest code has been rebased from base branch (usually `develop`)
- [ ]  Commits follow guidelines (concise, squashed, etc)
- [ ]  Github issues have been linked in relevant commits
- [ ]  Relevant reviewers (Senior Dev/EM/Designers) have been assigned to this PR

### Critical Changes

- Database change / migration to run
- Environment config change
- Breaking API change

### Related PRs

- #PRNUMBER

### How to Test

1. Please add or build upon a testing card to the [Notion page](http://notion.so) and link it here

### CODE REVIEW

**Comment by its-edmund** [comment]
Still needs #162 to be finished before this PR is done.

File: server/mongodb/actions/ResetUser.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import bcrypt from "bcryptjs";
+import jwt from "jsonwebtoken";
+import mongoDB from "../index";
+import PasswordResetRequest from "../models/PasswordResetRequest";
+import User from "../models/User";
+
+export async function resetUser(token, password) {
+  if (password == null) {
```
Comment (kphan32) [inline server/mongodb/actions/ResetUser.js:8]: The token should be validated as well

File: src/pages/api/users/resetpassword.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    const { token, password } = req.body;
+
+    return resetUser(token, password)
+      .then(() => {
+        res.status(200).json({ success: true });
+      })
+      .catch((error) =>
+        res.status(400).json({ success: false, payload: error.message })
+      );
+  }
```
Comment (kphan32) [inline src/pages/api/users/resetpassword.js:14]: Should send a response for other methods as well.

**Review by kphan32 — CHANGES_REQUESTED** [review]

File: server/mongodb/actions/ResetUser.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    throw new Error("Password must be provided!");
+  }
+
+  const hashedPassword = await bcrypt.hash(password, 10);
+
+  const request = await PasswordResetRequest.find({ token: token });
```
Comment (kphan32) [inline server/mongodb/actions/ResetUser.js:14]: Nit: use `findOne`

**Review by kphan32 — CHANGES_REQUESTED** [review]
Very close! Reminder to test changes before submitting for review.

**Comment by kphan32** [comment]
Also thought of this while looking at Luke's PR-- I think the actual endpoint to access this API should be `api/users?action=RESETPASSWORD` to maintain consistency. This is on me for not catching this earlier on, so feel free to resolve the previous changes and ping me so I can do this part.

File: server/mongodb/actions/ResetUser.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+  if (passwordRequest === null) {
+    throw new Error("Password reset query not found!");
+  }
+
+  const user = await User.findOneAndUpdate(
+    { email: passwordRequest.email },
+    { password: hashedPassword },
+    { new: false }
+  );
```
Comment (kphan32) [inline server/mongodb/actions/ResetUser.js:27]: nice

**Review by kphan32 — APPROVED** [review]
lgtm :+1:

**Pull request merged**